src/TechObject/Base/BaseOperation.cs

```diff
@@ -211,25 +211,30 @@ private void SetItems()
         public string SaveAsLuaTable(string prefix)
         {
             var res = "";
-            
-            if (Properties == null)
+            var propertiesCountForSave = Properties.Count();
+            if (Properties == null || propertiesCountForSave <= 0)
             {
                 return res;
             }
 
-            var propertiesCountForSave = Properties.Count();
-            if (propertiesCountForSave <= 0)
+            string paramsForSave = "";
+            foreach (var operParam in Properties)
             {
-                return res;
+                if(!operParam.IsEmpty)
+                {
+                    paramsForSave += "\t" + prefix + operParam.LuaName + 
+                        " = \'" + operParam.Value + "\',\n";
+                }           
             }
 
-            res += prefix + "props =\n" + prefix + "\t{\n";
-            foreach (var operParam in Properties)
+            bool needSavingParameters = paramsForSave != "";
+            if (needSavingParameters)
```

Comment:
Может такой вариант использовать:
```suggestion
            if (not String.IsNullOrEmpty(paramsForSave))
```

Reply:
Я против. Это ухудшает читабельность кода

Reply:
Сейчас спросим у прошедших курсы повышения недавно - @NickRisky - что скажешь?

Reply:
@idzm, рекомендую посмотреть книгу "Совершенный код", глава 12 пункт 5. Причем, этот прием используют и мои друзья-программисты. Но мнение @NickRisky тоже нужно выслушать 😃

Reply:
if (!string.IsNullOrEmpty(paramsForSave))
очень распостраненная проверка и я не считаю что она ухудшает читаемость, ведь многие программисты вкурсе про такое.

Reply:
Насчет клин-кода, оттуда стоит брать полезные советы, но сильно увлекаться я также не советую, иначе получиться вот такая лишняя сложность.

Reply:
@idzm, тогда надо использовать String.Empty, потому что туда никогда не придет значение null 😄